utils.py

```diff
@@ -214,10 +214,16 @@ def get_task_ip_and_ports(app, task):
             portMappings = app.get('container', {})\
                 .get('docker', {})\
                 .get('portMappings', [])
-            ports = filter(lambda p: p is not None,
-                           map(lambda p: p.get('containerPort', None),
-                               portMappings))
-            task_ports = list(ports)
+            if portMappings:
```

Comment:
```python
_port_mappings = portMappings or app.get('portDefinitions', [])
_attr = 'containerPort' if portMappings else 'port'
ports = [p.get(_attr, None) for p in _port_mappings if _attr in p]
```

It will remove the if